pkg/agent/manager/subscriber.go

```diff
@@ -3,30 +3,64 @@ package manager
 import (
 	"github.com/satori/go.uuid"
 	"github.com/spiffe/spire/pkg/agent/manager/cache"
+	"github.com/spiffe/spire/pkg/common/selector"
 )
 
 type subscriber struct {
-	c    chan []cache.Entry
+	c    chan *cache.Entry
 	sel  cache.Selectors
 	done chan struct{}
 }
 
-// subscribers is a map keyed by the Cache key, with a vaule mapped by Subscriber ID.
-type subscribers map[string]map[string]subscriber
+//type sID string
+// subscribers is a map keyed by the string representation of the selector sets, with a value mapped by Subscriber ID.
+// used to maintain a subscription of workloads
 
-func (s *subscribers) Add(sub subscriber) error {
+type subscribers struct {
+	selMap map[string][]uuid.UUID
+	sidMap map[uuid.UUID]*subscriber
```

Comment:
interesting, i didn't realize you could use non-string or int type as a map key... TIL